### ISSUE TITLE
Auto.* error:  The provided lr scheduler `StepLR` doesn't follow PyTorch's LRScheduler API

### ISSUE THREAD

**Issue by uumami** [Nixtla/nixtla#50]
Environment:
Installed from pip

Code:
```
nhits_config = {
       "max_steps": 100,                                                         # Number of SGD steps
       "input_size": 24,                                                         # Size of input window
       "learning_rate": tune.loguniform(1e-5, 1e-1),                             # Initial Learning rate
       "n_pool_kernel_size": tune.choice([[2, 2, 2], [16, 8, 1]]),               # MaxPool's Kernelsize
       "n_freq_downsample": tune.choice([[168, 24, 1], [24, 12, 1], [1, 1, 1]]), # Interpolation expressivity ratios
       "val_check_steps": 50,                                                    # Compute validation every 50 steps
       "random_seed": tune.randint(1, 10),                                       # Random seed
    }
model = AutoNHITS(h=12,
                  loss=SMAPE(),
                  config=nhits_config, 
                  search_alg=HyperOptSearch(),
                #hist_exog_list = ['lat', 'lon', 'speed', 'notification_id', 'ignition', 'cell'], # <- Historical exogenous variables
                  num_samples=20, gpus=1)

nf = NeuralForecast(models=[model], freq='S')
nf.fit(df=df_train, val_size=24)
```

Error: 
```
`2023-04-22 00:45:33,035	ERROR trial_runner.py:987 -- Trial train_tune_fb765440: Error processing event.
ray.exceptions.RayTaskError(MisconfigurationException): [36mray::ImplicitFunc.train()[39m (pid=24189, ip=172.17.0.2, repr=train_tune)
  File "/root/miniconda3/lib/python3.10/site-packages/ray/tune/trainable/trainable.py", line 349, in train
    result = self.step()
  File "/root/miniconda3/lib/python3.10/site-packages/ray/tune/trainable/function_trainable.py", line 417, in step
    self._report_thread_runner_error(block=True)
  File "/root/miniconda3/lib/python3.10/site-packages/ray/tune/trainable/function_trainable.py", line 589, in _report_thread_runner_error
    raise e
  File "/root/miniconda3/lib/python3.10/site-packages/ray/tune/trainable/function_trainable.py", line 289, in run
    self._entrypoint()
  File "/root/miniconda3/lib/python3.10/site-packages/ray/tune/trainable/function_trainable.py", line 362, in entrypoint
    return self._trainable_func(
  File "/root/miniconda3/lib/python3.10/site-packages/ray/tune/trainable/function_trainable.py", line 684, in _trainable_func
    output = fn()
  File "/root/miniconda3/lib/python3.10/site-packages/ray/tune/trainable/util.py", line 368, in _inner
    inner(config, checkpoint_dir=None)
  File "/root/miniconda3/lib/python3.10/site-packages/ray/tune/trainable/util.py", line 359, in inner
    trainable(config, **fn_kwargs)
  File "/root/miniconda3/lib/python3.10/site-packages/neuralforecast/common/_base_auto.py", line 28, in train_tune
    model.fit(dataset, val_size=val_size, test_size=test_size)
  File "/root/miniconda3/lib/python3.10/site-packages/neuralforecast/common/_base_windows.py", line 569, in fit
    trainer.fit(self, datamodule=datamodule)
  File "/root/miniconda3/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 770, in fit
    self._call_and_handle_interrupt(
  File "/root/miniconda3/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 723, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
  File "/root/miniconda3/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 811, in _fit_impl
    results = self._run(model, ckpt_path=self.ckpt_path)
  File "/root/miniconda3/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 1217, in _run
    self.strategy.setup(self)
  File "/root/miniconda3/lib/python3.10/site-packages/pytorch_lightning/strategies/single_device.py", line 72, in setup
    super().setup(trainer)
  File "/root/miniconda3/lib/python3.10/site-packages/pytorch_lightning/strategies/strategy.py", line 139, in setup
    self.setup_optimizers(trainer)
  File "/root/miniconda3/lib/python3.10/site-packages/pytorch_lightning/strategies/strategy.py", line 128, in setup_optimizers
    self.optimizers, self.lr_scheduler_configs, self.optimizer_frequencies = _init_optimizers_and_lr_schedulers(
  File "/root/miniconda3/lib/python3.10/site-packages/pytorch_lightning/core/optimizer.py", line 195, in _init_optimizers_and_lr_schedulers
    _validate_scheduler_api(lr_scheduler_configs, model)
  File "/root/miniconda3/lib/python3.10/site-packages/pytorch_lightning/core/optimizer.py", line 350, in _validate_scheduler_api
    raise MisconfigurationException(
pytorch_lightning.utilities.exceptions.MisconfigurationException: The provided lr scheduler `StepLR` doesn't follow PyTorch's LRScheduler API. You should override the `LightningModule.lr_scheduler_step` hook with your own logic if you are using a custom LR scheduler.`
```

**Comment by uumami** [comment]
The issue comes from pytorch_lighting [issue](https://github.com/Lightning-AI/lightning/issues/15912) updates.
I haven checked the conda version
There are three possible solutions:

+ Install dev-mode
  '''
  git clone https://github.com/Nixtla/neuralforecast.git
  cd neuralforecast
  pip install -e .
  '''

+ Freeze the version of pip of pytorch_lighting to 2.0.0 (you can keep pytorch 2.0)

`pip install neuralforecast 'pytorch_lightning==2.0.0'`

+ Freeze the version of pip of pytorch_lighting to <2.0 (you have to downgrade pytorch < 2.0)

`pip install neuralforecast 'pytorch_lightning==2.0.0'`
